api/controllers/externalLink.js

```diff
@@ -0,0 +1,330 @@
+					})
+					.catch((error) => {
+						defaultLog.error(error);
+						return Actions.sendResponse(res, 400, error);
+					});
+			})
+    .catch(error => defaultLog.error(error));
+  } catch (e) {
+    defaultLog.error(e);
+    // Delete the path details before we return to the caller.
+    delete e['path'];
+    return Actions.sendResponse(res, 500, e);
+  }
+};
+
+exports.protectedPublish = async (args, res) => {
+  defaultLog.info('EXTERNAL LINK PROTECTED PUBLISH');
+  const objId = args.swagger.params.exLinkId.value;
+  defaultLog.info("Publish External Link:", objId);
+
+  const ExternalLink = require('mongoose').model('ExternalLink');
+  try {
+    const exLink = await ExternalLink.findOne({ _id: objId });
+    if (exLink) {
+      defaultLog.info("External Link:", exLink);
+      const published = await Actions.publish(await exLink.save());
+      Utils.recordAction('Publish', 'ExternalLink', args.swagger.params.auth_payload.preferred_username, objId);
+      return Actions.sendResponse(res, 200, published);
+    } else {
+      defaultLog.info("Couldn't find that external link!");
+      return Actions.sendResponse(res, 404, e);
+    }
+  } catch (e) {
+    return Actions.sendResponse(res, 400, e);
+  }
+};
+
+exports.protectedUnPublish = async (args, res) => {
+  defaultLog.info('EXTERNAL LINK PROTECTED UNPUBLISH');
+  const objId = args.swagger.params.exLinkId.value;
+  defaultLog.info("Unpublish External Link:", objId);
+  const ExternalLink = require('mongoose').model('ExternalLink');
+  try {
+    const exLink = await ExternalLink.findOne({ _id: objId });
+    if (exLink) {
+      const unPublished = await Actions.unPublish(await exLink.save());
+      Utils.recordAction('Unpublish', 'ExternalLink', args.swagger.params.auth_payload.preferred_username, objId);
+      defaultLog.info("Published external link:", objId);
+      return Actions.sendResponse(res, 200, unPublished);
+    } else {
+      defaultLog.info("Couldn't find that external link!");
+      return Actions.sendResponse(res, 404, e);
+    }
+  } catch (e) {
+    defaultLog.error(e);
+    return Actions.sendResponse(res, 400, e);
+  }
+};
+
+exports.protectedPut = async (args, res) => {
+  defaultLog.info('EXTERNAL LINK PROTECTED PUT');
+  const objId = args.swagger.params.exLinkId.value;
+  let obj = {};
+  defaultLog.info('Put external link:', objId);
+
+  obj._updatedBy = args.swagger.params.auth_payload.preferred_username;
+  obj.displayName = args.swagger.params.displayName.value;
+	obj.externalLink = args.swagger.params.externalLink.value;
+  obj.section = args.swagger.params.section.value;
+  obj.projectPhase = args.swagger.params.projectPhase.value;
+  obj.dateAdded = args.swagger.params.dateAdded.value;
+  obj.dateUpdated = args.swagger.params.dateUpdated.value;
+  obj.description = args.swagger.params.description.value;
+  obj.section = "null" === args.swagger.params.section.value ? null : args.swagger.params.section.value;
```

Comment:
I think this would do the same thing. There may be an even shorter way to do it
```suggestion
  obj.section = args.swagger.params.section.value ?  args.swagger.params.section.value : null;
```

Reply:
Unfortunately I don't think it would work in this situation, since we're looking for a string with the text "null".

Reply:
You're right! I remember that sections can be "null" sometimes.

Reply:
Yeah I believe it's because everything has to be stringified for the API request and then converted back